community/contributing/pr_review_guidelines.rst

```diff
@@ -0,0 +1,335 @@
+  times to know how to make one, but for a general template
+  think about *"Fix <issue> in <part of codebase>"*.
+
+4. GitHub checklist
+^^^^^^^^^^^^^^^^^^^
+
+* **Validate the target branch of the PR.**
+
+  Godot development happens around the ``master`` branch. Therefore
+  most pull requests must be made against it, and can then be
+  backported to other branches. Be wary of people making PRs
+  on the version they are working on (e.g, ``3.3``) and guide
+  them to make a change against a higher-order branch. If the
+  change doesn't make sense for the ``master``, the initial PR can
+  be made against the current maintenance branch, such as ``3.x``.
+  It's okay for people to make multiple PRs for each target
+  branch, especially if the changes cannot be easily backported.
+  Cherry-picking is also an option, if possible. Use the appropriate
+  labels if the PR can be cherrypicked (e.g. ``cherrypick:3.x``).
+
+.. note::
+	It is possible to change the target branch of the PR,
+	that has already been submitted, but be aware of the
+	consequences. As it cannot be synchronized with the push,
+	the target branch change will inevitable tag the entire
+	list of maintainers for review. It may also render the
+	CI incapable of running properly. A push should help with
+	that, but if nothing else, recommend opening a new, fresh PR.
+
+* **Make sure that the appropriate milestone is assigned.**
+
+  This will make it more obvious which version would include the
+  submitted changes, should the pull request be merged now.
+  Note, that the milestone is not a binding contract and does
+  not guarantee that this version is definitely going to include
+  the PR. If the pull request is not merged before the version
+  is released, the milestone is moved (and the PR itself may
+  require a target branch change).
+
+* **Make sure that the opening message of the PR contains the
+  magic words "Closes #..." or "Fixes #...".**
+
+  These link the PR and the referenced issue together and allow
+  to auto-close the latter when you merge the changes. Note, that
+  this only works for the PRs that target the ``master`` branch.
+  For others you need to pay attention and close the related
+  issues manually. Do it with *"Fixed by #..."* or *"Resolved by #..."*
+  comment to clearly indicate the act for future generations.
+
+* **For the issues that get closed by the PR add the closest
+  relevant milestone.**
+
+  In other words, if the PR is targeting the ``master``, but is then
+  also cherrypicked for ``3.x``, the next ``3.x`` release would be the
+  appropriate milestone for the issue.
+
+5. Don't be afraid
+^^^^^^^^^^^^^^^^^^
+
+That's the beauty of version control systems — you can always
+revert, go back to a previous commit that was working. We use
+Git, so you shouldn't be afraid to merge good pull requests
+out of concerns that it may be a mistake. It can, but it's not
+a problem and nobody is going to judge you for it.
```

Comment:
We should mention something about while people shouldn't be afraid, they should also be confident about their decision. Reverting is always an option, but I think it's better if we don't have to because a merged change is very likely to be good so few of them need reverting.

Reply:
That's kind of at odds with what @akien-mga said, so I'm not sure. Of course you should be confident, but one of the points was that people who should already be confident may not be because they are afraid of a revert and critique for making a wrong decision. The idea is to destigmatize the chance to make a mistake.